WordPress/src/main/java/org/wordpress/android/ui/blaze/ui/blazeoverlay/BlazeOverlayFragment.kt

```diff
@@ -0,0 +1,282 @@
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.constraintlayout.compose.Dimension
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import coil.compose.rememberImagePainter
+import dagger.hilt.android.AndroidEntryPoint
+import org.wordpress.android.R
+import org.wordpress.android.ui.blaze.BlazeUiState
+import org.wordpress.android.ui.blaze.PostUIModel
+import org.wordpress.android.ui.compose.components.Drawable
+import org.wordpress.android.ui.compose.components.ImageButton
+import org.wordpress.android.ui.compose.theme.AppColor
+import org.wordpress.android.ui.compose.theme.AppTheme
+import org.wordpress.android.ui.compose.unit.FontSize
+import org.wordpress.android.ui.compose.unit.Margin
+import org.wordpress.android.ui.compose.utils.LocaleAwareComposable
+import org.wordpress.android.ui.utils.UiString
+import org.wordpress.android.util.LocaleManager
+
+@AndroidEntryPoint
+class BlazeOverlayFragment : Fragment() {
+    companion object {
+        fun newInstance() = BlazeOverlayFragment()
+    }
+
+    private val viewModel: BlazeViewModel by activityViewModels()
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View = ComposeView(requireContext()).apply {
+        setContent {
+            AppTheme {
+                val userLanguage by viewModel.refreshAppLanguage.observeAsState("")
+                val postModel by viewModel.promoteUiState.observeAsState(BlazeUiState.PromoteScreen.Site)
+                LocaleAwareComposable(
```

Comment:
I probably should have added what I'm about to say here in the documentation of `LocaleAwareComposable`:

It should not be used unless the host view is **not** recreated on purpose when the locale changes while the screen is visible.

If `LocaleAwareActivity` can be used it should be instead of `LocaleAwareComposable`, the composable is only for those particular edge cases when the updated locale isn't propagated to the `ComposeView` and by default it is when we use `LocaleAwareActivity`.

As discussed also via DM, if it changing the language works without `LocaleAwareComposable` I advice to remove it and the related code, making things much simpler.

I tried to test locally if without it there are locale inconsistencies, but for some reason the **Blaze** dashboard card gets hidden when I switch to another language.

Reply:
@ovitrif 

> I tried to test locally if without it there are locale inconsistencies, but for some reason the Blaze dashboard card gets hidden when I switch to another language.

This is the correct behavior, as we are only supporting English languages for now.

Reply:
I have removed the LocaleAwareComposable with the commit 4b4d04f7550fda5cafa3b50a10d372ac0d6337e7